core/src/main/java/org/apache/spark/shuffle/sort/io/LocalDiskShuffleMapOutputWriter.java

```diff
@@ -87,7 +86,7 @@ public ShufflePartitionWriter getPartitionWriter(int reducePartitionId) throws I
     }
     lastPartitionId = reducePartitionId;
     if (outputTempFile == null) {
-      outputTempFile = Utils.tempFileWith(outputFile);
+      outputTempFile = blockResolver.createTempFile(outputFile);
```

Comment:
I might be missing some context here - why do we need to do this ?
In order to delete from shuffle service, we only need to ensure the directory hosting the block has read/write/execute acl for shuffle service, right ?

Reply:
Some of the context is here: https://github.com/apache/spark/pull/35085#discussion_r788725899

Basically in a secure yarn environment, the `setgid` bit is set on all directories so that it keeps the same group as the node manager and shuffle service, but when we add the group write permission to the directory, it removes the setgid bit, so shuffle files get created without the node manager group with mode `rw-r-----`, and the shuffle service doesn't have permission to read it. This creates the shuffle files world readable to the shuffle service can still _read_ the files with this update

Reply:
That makes sense - given this, why not simply use `DiskBlockManager.createDirWithPermission770` while creating the directory and avoid needing to set any permission ?

Reply:
Missed this and not sure if this was covered in a different thread, but `mkdir -p770` also loses the setgid bit